ext/intl/dateformat/datetimepatterngenerator_methods.cpp

```
@@ -0,0 +1,153 @@
+				"datetimepatterngenerator_create: error creating DateTimePatternGenerator",
+				0);
+		goto error;
+	}
+
+error:
+	return U_FAILURE(intl_error_get_code(NULL)) ? FAILURE : SUCCESS;
+}
+/* }}} */
+
+/* {{{ Create IntlDateTimePatternGenerator. */
+U_CFUNC PHP_FUNCTION( datetimepatterngenerator_create )
+{
+    object_init_ex( return_value, IntlDateTimePatternGenerator_ce_ptr );
+    if (dtpg_ctor(INTERNAL_FUNCTION_PARAM_PASSTHRU) == FAILURE) {
+		zval_ptr_dtor(return_value);
+		RETURN_NULL();
+	}
+}
+/* }}} */
+
+/* {{{ IntlDateTimePatternGenerator object constructor. */
+U_CFUNC PHP_METHOD( IntlDateTimePatternGenerator, __construct )
+{
+	zend_error_handling error_handling;
+
+	zend_replace_error_handling(EH_THROW, IntlException_ce_ptr, &error_handling);
+	/* return_value param is being changed, therefore we will always return
+	 * NULL here */
+	return_value = ZEND_THIS;
+	if (dtpg_ctor(INTERNAL_FUNCTION_PARAM_PASSTHRU) == FAILURE) {
+		if (!EG(exception)) {
+			zend_string *err = intl_error_get_message(NULL);
+			zend_throw_exception(IntlException_ce_ptr, ZSTR_VAL(err), intl_error_get_code(NULL));
+			zend_string_release_ex(err, 0);
+		}
+	}
+	zend_restore_error_handling(&error_handling);
+}
+/* }}} */
+
+
+/* {{{ Get date format pattern from skeleton. */
+U_CFUNC PHP_FUNCTION(datetimepatterngenerator_get_best_pattern)
+{
+	char			*skeleton_str	= NULL;
+	size_t			skeleton_len;
+	UnicodeString	skeleton_uncleaned;
+	UnicodeString	skeleton;
+
+	DTPATTERNGEN_METHOD_INIT_VARS;
+
+	/* Parse parameters. */
+	if( zend_parse_method_parameters( ZEND_NUM_ARGS(), getThis(), "Os",
+		&object, IntlDateTimePatternGenerator_ce_ptr,  &skeleton_str, &skeleton_len ) == FAILURE )
+	{
+		RETURN_THROWS();
+	}
+
+	DTPATTERNGEN_METHOD_FETCH_OBJECT;
+
+	intl_stringFromChar(skeleton_uncleaned, skeleton_str, skeleton_len, DTPATTERNGEN_ERROR_CODE_P(dtpgo));
+
+	INTL_METHOD_CHECK_STATUS(dtpgo, "datetimepatterngenerator_get_best_pattern: skeleton was not a valid UTF-8 string");
```

```suggestion
	INTL_METHOD_CHECK_STATUS(dtpgo, "Skeleton is not a valid UTF-8 string");
```

So the function name shouldn't be added (it's superfluous), the message should start with a capital character, and AFAIR we don't use past tense

---

Also, I'm wondering if we can consider an invalid UTF-8 skeleton a programmatic error? As far as I see, we can, because the input should come from the programmer, not from the user. If my hypothesis is ok, then we could throw an exception instead of these warnings, like this:
`zend_argument_value_error(2, "must be a valid UTF-8 string")` (the `2` should be adapted in case of the method counterpart, since `$skeleton` is the 1st parameter there).

Last year, we migrated hundreds of warnings to exceptions, so it would be nice to follow this practice here as well, if possible.

---

I have adjusted the messages.
I would agree that this is most likely a programming error - I used a warning here for consistency with similar cases (eg. `IntlDateFormatter::setPattern()`), but I can switch to using an exception if preferred.